nym-api/src/support/storage/manager.rs

```diff
@@ -1329,6 +1386,146 @@ impl StorageManager {
         .fetch_all(&self.connection_pool)
         .await
     }
+
+    /// Fetches raw route results from the database for a given time interval.
+    /// Assumes the 'routes' table has layer1, layer2, layer3, gw, success, and a timestamp.
+    async fn get_raw_routes_in_interval(
+        &self,
+        start_ts_secs: i64,
+        end_ts_secs: i64,
+    ) -> Result<Vec<(NodeId, NodeId, NodeId, NodeId, bool)>, sqlx::Error> {
+        // Temporary struct to match the expected columns from the 'routes' table
+        // NodeId here is assumed to be compatible with how layer1, etc. are stored (e.g. u32/i32/i64)
+        struct RawRouteData {
+            layer1: i64,
+            layer2: i64,
+            layer3: i64,
+            gw: i64,
+            success: Option<bool>,
+            // timestamp: i64, // Not explicitly selected into struct, but used in WHERE and ORDER BY
+        }
+
+        // Ensure your 'routes' table has:
+        // layer1 (NodeId type), layer2 (NodeId type), layer3 (NodeId type),
+        // gw (NodeId type), success (bool/INTEGER), timestamp (BIGINT/INTEGER)
+        // The "NodeId:" type hint for sqlx::query_as! helps map to nym_mixnet_contract_common::NodeId
+        let db_routes = sqlx::query_as!(
+            RawRouteData,
+            r#"
+            SELECT
+                layer1 as "layer1",
+                layer2 as "layer2",
+                layer3 as "layer3",
+                gw as "gw",
+                success
+            FROM routes
+            WHERE timestamp >= ? AND timestamp <= ?
+            ORDER BY timestamp ASC
+            "#,
+            start_ts_secs,
+            end_ts_secs
+        )
+        .fetch_all(&self.connection_pool)
+        .await?;
+
+        Ok(db_routes
+            .into_iter()
+            .map(|r| {
+                (
+                    r.layer1 as NodeId,
+                    r.layer2 as NodeId,
+                    r.layer3 as NodeId,
+                    r.gw as NodeId,
+                    r.success.unwrap_or_default(),
```

Comment:
why would success be missing? can't we set it to `NOT NULL` and then avoid this whole iterator?

Reply:
I don't know, just being paranoid, if we set it to NOT NULL then inserts might start failing quietly if we have a problem somewhere else, better be explicit then sorry :)